app/src/main/java/org/mozilla/fenix/components/toolbar/CFRPresenter.kt

```diff
@@ -0,0 +1,130 @@
+    internal var tcpCfrScope: CoroutineScope? = null
+    @VisibleForTesting
+    internal var tcpCfr: CFRPopup? = null
+
+    /**
+     * Start observing [store] for updates which may trigger showing a CFR.
+     */
+    @Suppress("MagicNumber")
+    fun start() {
+        if (settings.shouldShowTotalCookieProtectionCFR) {
+            tcpCfrScope = store.flowScoped { flow ->
+                flow
+                    .mapNotNull { it.findCustomTabOrSelectedTab(sessionId)?.content?.progress }
+                    // The "transformWhile" below ensures that the 100% progress is only collected once.
+                    .transformWhile { progress ->
+                        emit(progress)
+                        progress != 100
+                    }
+                    .filter { it == 100 }
+                    .collect {
+                        tcpCfrScope?.cancel()
+                        showTcpCfr()
+                    }
+            }
+        }
+    }
+
+    /**
+     * Stop listening for [store] updates.
+     * CFRs already shown are not automatically dismissed.
+     */
+    fun stop() {
+        tcpCfrScope?.cancel()
+    }
+
+    @VisibleForTesting
+    internal fun showTcpCfr() {
+        CFRPopup(
+            text = context.getString(R.string.tcp_cfr_message),
+            anchor = toolbar.findViewById(
+                R.id.mozac_browser_toolbar_security_indicator
+            ),
+            properties = CFRPopupProperties(
+                indicatorDirection = if (settings.toolbarPosition == ToolbarPosition.TOP) {
+                    CFRPopup.IndicatorDirection.UP
+                } else {
+                    CFRPopup.IndicatorDirection.DOWN
+                },
+            ),
+            onDismiss = {
+                when (it) {
+                    true -> TrackingProtection.tcpCfrExplicitDismissal.record(NoExtras())
+                    false -> TrackingProtection.tcpCfrImplicitDismissal.record(NoExtras())
+                }
+            }
+        ) {
+            Text(
+                text = context.getString(R.string.tcp_cfr_learn_more),
+                color = FirefoxTheme.colors.textOnColorPrimary,
+                modifier = Modifier.clickable {
+                    context.components.useCases.tabsUseCases.selectOrAddTab.invoke(
+                        SupportUtils.getSumoURLForTopic(
+                            context,
+                            TOTAL_COOKIE_PROTECTION
+                        )
+                    )
+                    TrackingProtection.tcpSumoLinkClicked.record(NoExtras())
+                    tcpCfr?.dismiss()
+                },
+                style = FirefoxTheme.typography.body2.copy(
+                    textDecoration = TextDecoration.Underline
+                )
+            )
+        }.apply {
+            settings.shouldShowTotalCookieProtectionCFR = false
+            tcpCfr = this
+            show()
+            TrackingProtection.tcpCfrShown.record(NoExtras())
```

Comment:
Nit:
May be we could move this function into the body of `fun showTcpCfr()`? `apply` is usually used for setting parameters or at list calls on `this`. I see why one could put an analytics call there, but it looks a bit out of place in there.

Reply:
Can use `run` instead of `apply`.
I like how there is one block of code for initializing the popup and then one for any other business logic.